app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -229,93 +194,35 @@ export default Component.extend(KeyEnterEscape, {
     const { isIOS, isAndroid, isOpera } = this.capabilities;
     const showAtEnd = isMobileDevice || isIOS || isAndroid || isOpera;
 
-    const boundaryPosition = this._getRangeBoundaryRect(firstRange, showAtEnd);
+    if (showAtEnd) {
+      this.placement = "bottom-start";
+    }
 
     // change the position of the button
     schedule("afterRender", () => {
       if (!this.element || this.isDestroying || this.isDestroyed) {
         return;
       }
 
-      let top = 0;
-      let left = 0;
-      const pxFromSelection = 5;
-
-      if (showAtEnd) {
-        // The selection-handles on iOS have a hit area of ~50px radius
-        // so we need to make sure our buttons are outside that radius
-        // Apply the same logic on all mobile devices for consistency
-
-        top = boundaryPosition.bottom + pxFromSelection;
-        left = boundaryPosition.left;
-
-        const safeRadius = 50;
+      this.textRange = new VirtualElementFromTextRange(".cooked");
 
-        const topicArea = document
-          .querySelector(".topic-area")
-          .getBoundingClientRect();
-        topicArea.x += document.documentElement.scrollLeft;
-        topicArea.y += document.documentElement.scrollTop;
-
-        const endHandlePosition = boundaryPosition;
-        const width = this.element.clientWidth;
-
-        const possiblePositions = [
+      this._popper = createPopper(this.textRange, this.element, {
```

Comment:
`this._popper` needs to be initialized as well, and then destroyed via `this._popper?.destroy()`.